NICE.Identity.Management/ClientApp/src/views/Website/Website.tsx

```diff
@@ -0,0 +1,395 @@
+			pageNumber = 1;
+		}
+
+		return pageNumber;
+	};
+
+	getPaginationText = (
+		usersCount: number,
+		start: number,
+		finish: number,
+	): string => {
+		const amountPerPage = finish - start;
+		const paginationExtract =
+		usersCount > amountPerPage ? `${start + 1} to ${finish} of ` : "";
+
+		return `Showing ${paginationExtract}${usersCount} user${
+			usersCount === 1 ? "" : "s"
+		}`;
+	};
+
+	getPaginateStartAndFinishPosition = (
+		userCount: number,
+		pageNumber: number,
+		itemsPerPage: number | string,
+	): { start: number; finish: number } => {
+		const paginationPositions = {
+			start: 0,
+			finish: userCount,
+		};
+
+		const itemAmountIsNumber = Number(itemsPerPage);
+
+		if (itemAmountIsNumber) {
+			paginationPositions.start = (pageNumber - 1) * itemAmountIsNumber;
+			paginationPositions.finish =
+				paginationPositions.start + itemAmountIsNumber <= userCount
+					? paginationPositions.start + itemAmountIsNumber
+					: userCount;
+		}
+
+		return paginationPositions;
+	};
+
+	changeAmount = (e: React.ChangeEvent<HTMLSelectElement>): void => {
+		let itemsPerPage = (e.target as HTMLSelectElement).value || 25,
+			pageNumber = this.state.pageNumber,
+			path = stripMultipleQueries(this.state.path, ["amount", "page"]);
+
+		itemsPerPage = Number(itemsPerPage) ? Number(itemsPerPage) : itemsPerPage;
+
+		pageNumber = this.pastPageRange(
+			itemsPerPage,
+			pageNumber,
+			this.state.usersAndRoles.length,
+		);
+
+		path = appendQueryParameter(path, "amount", itemsPerPage.toString());
+		path = appendQueryParameter(path, "page", pageNumber);
+
+		this.setState({ itemsPerPage, path, pageNumber }, () => {
+			this.props.history.push(path);
+		});
+	};
+
+	changePage = (e: React.MouseEvent<HTMLAnchorElement>): void => {
+		e.preventDefault();
+
+		let pageNumber =
+				(e.target as HTMLAnchorElement).getAttribute("data-pager") || 1,
+			path = removeQueryParameter(this.state.path, "page");
+
+		if (pageNumber === "previous") {
+			pageNumber = this.state.pageNumber - 1;
+		}
+		if (pageNumber === "next") {
+			pageNumber = this.state.pageNumber + 1;
+		}
+
+		pageNumber = Number(pageNumber);
+		path = appendQueryParameter(path, "page", pageNumber);
+
+		this.setState({ pageNumber, path }, () => {
+			this.props.history.push(path);
+		});
+	};
+
+    render(): JSX.Element {
+		const { websiteUsersAndRoles, usersAndRoles, error, isLoading, pageNumber, itemsPerPage, } = this.state;
+
+		const lastBreadcrumb = isLoading ? "Loading service details" : websiteUsersAndRoles.website?.service?.name;
+
+		const paginationPositions = this.getPaginateStartAndFinishPosition(
+			usersAndRoles?.length,
+			pageNumber,
+			itemsPerPage,
+		);
+
+		const paginationText = this.getPaginationText(
+			usersAndRoles?.length,
+			paginationPositions.start,
+			paginationPositions.finish,
+		);
+
+		const usersPaginated = usersAndRoles?.length
+		? usersAndRoles.slice(paginationPositions.start, paginationPositions.finish)
+		: usersAndRoles;
```

Comment:
Should this be renamed usersAndRolesPaginated? I assumed so and included it in my suggestion below (for the table).